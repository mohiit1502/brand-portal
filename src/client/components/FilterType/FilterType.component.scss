@import "./../../styles/override/variables";

.c-FilterType {
  border: 1px solid $blue;
  border-radius: 2px;
  display: inline-block;
  color: $blue;
  background: #f3f9fc;
  padding-top: 2px;
  padding-bottom: 2px;

  .c-FilterType__closeButton {
    color: $blue;
    cursor: pointer;

    &:hover {
      color: $navy-blue;
      text-decoration: underline;
    }
  }
}
