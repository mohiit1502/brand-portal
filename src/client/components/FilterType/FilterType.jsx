import React from "react";
import PropTypes from "prop-types";
import "./FilterType.component.scss";

const FilterType = props => {

  // const clearFilter = () => {
  //   props.clearFilterHandler({...props.currentFilters, [props.filterId]: ""})
  // }

  const filters = props.filters

  const getFilterText = (filterName, filterOption) => {
    if (filterName === "Submitted By") {
      return filterName + " " + filterOption.name;
    } else if (filterName === "Associated Brands") {
      return "Associated Brand is " + filterOption.name;
    } else {
      return `${filterName} is ${filterOption.name}`;
    }
  }

  return filters.map(filter => {
    const filterId = filter.id;
    const filterName = filter.name
    const filterOptions = filter.filterOptions;
    const selectedFilterOption = filterOptions.filter(option => option.selected)
    if (selectedFilterOption.length > 0) {
      const allSelectedFilterOption = selectedFilterOption.filter(filterOption => filterOption.value === "all");
      if (allSelectedFilterOption.length <= 0) {
        return selectedFilterOption.map(filterOption => (
          <div className="c-FilterType font-size-14 mt-2 mr-1 px-2">
              <span className="c-FilterType__primaryText mr-3 line-height-reset">
                {getFilterText(filterName, filterOption)}
              </span>
            <span className="c-FilterType__closeButton line-height-reset font-size-20" onClick={() => props.clearFilter(filterId,filterOption.id)}>x</span>
          </div>)
        )
      }
    }
  });
};

FilterType.propTypes = {
  clearFilterHandler: PropTypes.func,
  currentFilters: PropTypes.object,
  filterMap: PropTypes.object,
  filterText: PropTypes.string
};

export default FilterType;
