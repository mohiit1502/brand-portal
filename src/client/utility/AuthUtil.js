import authConfig from "./../config/authorizations";
import Helper from "./helper";

export default class AuthUtil {
  static isActionAccessible (action, userProfile, actedOnRole) {
    if (action && action.indexOf(".") > -1) {
      const actionConfig = Helper.search(action, authConfig);
      if (actionConfig && userProfile && userProfile.role && userProfile.role.name && typeof actionConfig === "object") {
        const role = userProfile.role.name;
        if (actionConfig.length !== undefined) {
          return actionConfig.includes(role);
        } else {
          if (Object.keys(actionConfig).includes(role) && actedOnRole && actionConfig[role].includes(actedOnRole)) {
            return true;
          }
        }
      }
    }
    return false;
  }
}
