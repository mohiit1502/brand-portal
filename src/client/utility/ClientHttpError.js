export default class ClientHttpError {

  status;
  error;
  message;

  constructor (status, error, message) {
    this.status = status;
    this.error = error;
    this.message = message;
  }

}
