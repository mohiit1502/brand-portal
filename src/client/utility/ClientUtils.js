import queryString from "query-string";

export default class ClientUtils {

  static getQueryParams (location) {
    return queryString.parse(location.search);
  }
}
