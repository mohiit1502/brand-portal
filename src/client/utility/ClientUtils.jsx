import React from "react";
import queryString from "query-string";
import {Redirect} from "react-router";

export default class ClientUtils {

  static getQueryParams (location) {
    return queryString.parse(location.search);
  }

  static  getAllValuesFromRecursiveTree(tree) {

    if (typeof tree !== "object" || tree === null) {
      return [tree];
    }

    const list = [];

    for (const i in tree) {
      if (tree.hasOwnProperty(i)) {
        list.push(...this.getAllValuesFromRecursiveTree(tree[i]));
      }
    }
    return list;
  }

  static where (arrayList, filterObj) {
    let index = -1;
    for (let i = 0; i < arrayList.length; i++) {
      for (const j in filterObj) {
        if (filterObj.hasOwnProperty(j) && arrayList[i].hasOwnProperty(j) && filterObj[j] === arrayList[i][j]) {
          index = i;
        } else {
          index = -1;
        }
      }
      if (index !== -1) {
        return index;
      }
    }
    return index;
  }

}
