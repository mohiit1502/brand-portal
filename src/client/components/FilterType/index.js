import FilterType from "./FilterType.jsx";

export default FilterType;
