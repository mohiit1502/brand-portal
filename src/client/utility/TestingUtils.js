import {applyMiddleware, compose, createStore} from "redux";
import rootReducer from "../reducers";
import thunk from "redux-thunk";
import Immutable from "immutable";
import configureMockStore from 'redux-mock-store';
import ClientHttpError from "./ClientHttpError";

export const findByTestAttribute = (component,attr) => component.find(`[data-test="${attr}"]`);

export const testStore = state => configureMockStore()(state);

export const clearKeys = (tree, arr) => {
  if (arr && arr.indexOf(tree) > -1) {
    return;
  }
  if (tree && typeof tree === 'object') {
    if (tree.inputId) {
      return;
    }
    arr.push(tree)
    if (tree.key) {
      try {
        delete tree.key;
      } catch (e) {}
    }
    if (Array.isArray(tree)) {
      tree.forEach(obj => clearKeys(obj, arr));
    } else {
      Object.keys(tree).forEach(key => clearKeys(tree[key], arr));
    }
  }
}

export function setupFetchStub(data) {
  return function fetchStub(_url) {
    return new Promise((resolve) => {
      resolve({
        json: () =>
          Promise.resolve({
            data,
          }),
      })
    })
  }
}

export function setupFetchThrowStub() {
  const error = new ClientHttpError(500, "test-error"); // Directly throwing new ClientHttpError fails in Jest
  throw error;
}
