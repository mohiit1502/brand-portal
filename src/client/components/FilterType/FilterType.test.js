import React, {useRef} from "react";
import FilterType from "./FilterType";
import toJson from "enzyme-to-json";
import {mount} from "enzyme";
import {testStore} from "../../utility/TestingUtils";


let props;
let store;

const setUp = (props) => {
  store = testStore({});
  return mount(props ? <FilterType {...props} />: <FilterType />);
};

jest.mock("react", () => {
  const originReact = jest.requireActual("react");
  const mUseRef = jest.fn();
  return {
    ...originReact,
    useRef: mUseRef
  };
});



describe("FilterType", () => {
  let wrapper;
  beforeEach(() => {
    jest.restoreAllMocks();
    jest.resetAllMocks();
  });
  it("renders without error", () => {
    props = {
      clearFilterHandler: () => {},
      currentFilters: {},
      filterMap: {},
      filterText: "",
      filters:[
        {
          id: "1",
          name: "Submitted By",
          filterOptions: [{}, {}]
        },
        {
          id: "2",
          name: "Associated Brands",
          filterOptions: [{}, {}]
        },
        {
          id: "3",
          name: "test",
          filterOptions: [{}, {}]
        }
      ]
    };
    wrapper = setUp(props);

    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();

  });
});
