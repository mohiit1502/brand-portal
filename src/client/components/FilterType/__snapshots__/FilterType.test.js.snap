// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterType renders without error 1`] = `
<FilterType
  clearFilterHandler={[Function]}
  currentFilters={Object {}}
  filterMap={Object {}}
  filterText=""
  filters={
    Array [
      Object {
        "filterOptions": Array [
          Object {},
          Object {},
        ],
        "id": "1",
        "name": "Submitted By",
      },
      Object {
        "filterOptions": Array [
          Object {},
          Object {},
        ],
        "id": "2",
        "name": "Associated Brands",
      },
      Object {
        "filterOptions": Array [
          Object {},
          Object {},
        ],
        "id": "3",
        "name": "test",
      },
    ]
  }
/>
`;
